src/main/scala/com/gu/fastly/Lambda.scala

```diff
@@ -62,10 +65,39 @@ class Lambda {
       }
     }
 
+    // Republish events for successful deletes and updates as
+    // com.gu.crier.model.event.v1.Event events thrift serialized and base64 encoded
+    successfulPurges.foreach { event =>
+      val supportedDecacheEventType = event.eventType match {
+        case EventType.Update => Some(com.gu.fastly.model.event.v1.EventType.Update)
+        case EventType.Delete => Some(com.gu.fastly.model.event.v1.EventType.Delete)
+        case _ => None
+      }
+
+      supportedDecacheEventType.map { decacheEventType =>
+        val contentDecachedEvent = com.gu.fastly.model.event.v1.ContentDecachedEvent(
+          contentId = event.payloadId,
+          eventType = decacheEventType
+        )
+        try {
+          val message = ContentDecachedEventSerializer.serialize(contentDecachedEvent)
+          println("Publishing SNS decached message for content id '" + event.payloadId + "' of length: " + message.length)
+
+          val publishRequest = new PublishRequest()
+          publishRequest.setTopicArn(config.decachedContentTopic)
+          publishRequest.setMessage(message)
+          snsClient.publish(publishRequest)
+        } catch {
+          case t: Throwable =>
+            println("Warning; publish sns decached event failed: " + t.getMessage)
```

Comment:
as above - this would be prettier as `s"warning...event failed: ${t.getMessage}"`

Reply:
Yep - that's fair. Picking this up.